MapboxDirections/MBRouteOptions.swift

```diff
@@ -478,20 +478,21 @@ open class RouteOptions: NSObject, NSSecureCoding, NSCopying{
      - returns: A tuple containing an array of waypoints and an array of routes.
      */
     internal func response(from json: JSONDictionary) -> ([Waypoint]?, [Route]?) {
-        var namedWaypoints: [Waypoint]
-        if let waypoints = (json["waypoints"] as? [JSONDictionary]) {
-            namedWaypoints = zip(waypoints, self.waypoints).map { (json, waypoint) -> Waypoint in
-                let location = json["location"] as! [Double]
+        var namedWaypoints: [Waypoint]?
+        if let jsonWaypoints = (json["waypoints"] as? [JSONDictionary]) {
+            namedWaypoints = zip(jsonWaypoints, waypoints).map { (api, local) -> Waypoint in
+                let location = api["location"] as! [Double]
                 let coordinate = CLLocationCoordinate2D(geoJSON: location)
-                return Waypoint(coordinate: coordinate, name: waypoint.name ?? json["name"] as? String)
+                return Waypoint(coordinate: coordinate, name: local.name ?? api["name"] as? String)
             }
-        } else {
-            namedWaypoints = self.waypoints
         }
+        
+        let wpts = namedWaypoints ?? waypoints
```

Comment:
FYI, we try not to overly abbreviate like this in the codebase. In this case, we could instead inline the variable in the two lines below, or we could’ve overridden namedWaypoints and used that variable below.